pkg/cmd/roachtest/libpq.go

```diff
@@ -81,24 +82,52 @@ func registerLibPQ(r *testRegistry) {
 
 		_ = c.RunE(ctx, node, fmt.Sprintf("mkdir -p %s", resultsDir))
 
-		blocklistName, expectedFailures, ignorelistName, ignoredFailures := libPQBlocklists.getLists(version)
+		blocklistName, expectedFailures, ignorelistName, ignoreList := libPQBlocklists.getLists(version)
 		if expectedFailures == nil {
 			t.Fatalf("No lib/pq blocklist defined for cockroach version %s", version)
 		}
 		c.l.Printf("Running cockroach version %s, using blocklist %s, using ignorelist %s", version, blocklistName, ignorelistName)
 
 		t.Status("running lib/pq test suite and collecting results")
 
+		// List all the tests that start with Test or Example.
+		testListRegex := `"(Test|Example)"`
+		buf, err := c.RunWithBuffer(
+			ctx,
+			t.l,
+			node,
+			fmt.Sprintf("cd %s && PGPORT=26257 PGUSER=root PGSSLMODE=disable PGDATABASE=postgres go test -list %s", libPQPath, testListRegex),
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Convert the output of go test -list into an list.
+		tests := strings.Fields(string(buf))
+		var allowedTests []string
+
+		// The last line of output from go test -list Test is something like
+		// "ok  	github.com/lib/pq	0.003s", we ignore this by slicing the last entry.
+		for _, testName := range tests[:len(tests)-1] {
```

Comment:
just to be sure we don't miss anything, maybe we could loop over the whole slice, but then add a check like `if testName does not start with "Test"`